kartothek/io/dask/delayed.py

```diff
@@ -582,3 +585,45 @@ def store_delayed_as_dataset(
         metadata_merger=metadata_merger,
         metadata_storage_format=metadata_storage_format,
     )
+
+
+@default_docs
+def build_dataset_indices__delayed(
```

Comment:
Not a strong opinion but we could, ofc, add _better_ names, e.g. `build_secondary_indices__delayed`

We (I) shouldn't have put the `dataset` part in every function name :)

Reply:
It's actually aligned with the eager version. If we rename it here, we should also rename the other.

Reply:
I know it's aligned and you are right. Let's keep it as is and do a proper global sane renaming once we do a next major release (maybe)